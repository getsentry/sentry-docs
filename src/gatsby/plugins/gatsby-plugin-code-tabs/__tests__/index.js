const Remark = require("remark");

const plugin = require("../");

const remark = new Remark().data("settings", {
  commonmark: true,
  footnotes: true,
  pedantic: true,
});

describe("gatsby-plugin-code-tabs", () => {
  it("folds two code blocks", () => {
    const markdownAST = remark.parse(`
~~~python
print "Hello World!"
~~~

~~~javascript
console.log("Hello World!");
~~~
`);
    const transformed = plugin({ markdownAST }, {});
    expect(transformed).toMatchSnapshot();
  });

  it("does not fold code blocks split by a paragraph", () => {
    const markdownAST = remark.parse(`
~~~python
print "Hello World!"
~~~

some text here:

~~~javascript
console.log("Hello World!");
~~~
`);
    const transformed = plugin({ markdownAST }, {});
    expect(transformed).toMatchSnapshot();
  });

  it("does not fold code blocks on different levels", () => {
    const markdownAST = remark.parse(`
1. a list here
   ~~~plain
   inside list
   ~~~

~~~plain
outside list
~~~
`);
    const transformed = plugin({ markdownAST }, {});
    expect(transformed).toMatchSnapshot();
  });

  it("supports tab titles", () => {
    const markdownAST = remark.parse(`
~~~plain {tabTitle: Hello}
first
~~~

~~~plain {tabTitle: Goodbye}
second
~~~
`);
    const transformed = plugin({ markdownAST }, {});
    expect(transformed).toMatchSnapshot();
  });

  it("supports filenames", () => {
    const markdownAST = remark.parse(`
~~~plain {tabTitle: Hello} {filename: hello.txt}
first
~~~

~~~plain {tabTitle: Goodbye} {filename: goodbye.txt}
second
~~~
`);
    const transformed = plugin({ markdownAST }, {});
    expect(transformed).toMatchSnapshot();
  });
});
