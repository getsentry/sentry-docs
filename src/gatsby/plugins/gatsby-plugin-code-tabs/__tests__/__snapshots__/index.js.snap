// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gatsby-plugin-code-tabs does not fold code blocks on different levels 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [
        Object {
          "checked": null,
          "children": Array [
            Object {
              "children": Array [
                Object {
                  "position": Position {
                    "end": Object {
                      "column": 15,
                      "line": 2,
                      "offset": 15,
                    },
                    "indent": Array [],
                    "start": Object {
                      "column": 4,
                      "line": 2,
                      "offset": 4,
                    },
                  },
                  "type": "text",
                  "value": "a list here",
                },
              ],
              "position": Position {
                "end": Object {
                  "column": 15,
                  "line": 2,
                  "offset": 15,
                },
                "indent": Array [],
                "start": Object {
                  "column": 4,
                  "line": 2,
                  "offset": 4,
                },
              },
              "type": "paragraph",
            },
            Object {
              "children": Array [
                Object {
                  "type": "jsx",
                  "value": "<CodeTabs>",
                },
                Object {
                  "type": "jsx",
                  "value": "<CodeBlock language=\\"plain\\" title=\\"\\" filename=\\"\\">",
                },
                Object {
                  "lang": "plain",
                  "meta": null,
                  "position": Position {
                    "end": Object {
                      "column": 7,
                      "line": 5,
                      "offset": 49,
                    },
                    "indent": Array [
                      4,
                      4,
                    ],
                    "start": Object {
                      "column": 4,
                      "line": 3,
                      "offset": 19,
                    },
                  },
                  "type": "code",
                  "value": "inside list",
                },
                Object {
                  "type": "jsx",
                  "value": "</CodeBlock>",
                },
                Object {
                  "type": "jsx",
                  "value": "</CodeTabs>",
                },
              ],
              "data": Object {
                "hName": "div",
                "hProperties": Object {
                  "className": "code-tabs-wrapper",
                },
              },
              "lang": "plain",
              "meta": null,
              "position": Position {
                "end": Object {
                  "column": 7,
                  "line": 5,
                  "offset": 49,
                },
                "indent": Array [
                  4,
                  4,
                ],
                "start": Object {
                  "column": 4,
                  "line": 3,
                  "offset": 19,
                },
              },
              "type": "element",
              "value": "inside list",
            },
          ],
          "position": Position {
            "end": Object {
              "column": 7,
              "line": 5,
              "offset": 49,
            },
            "indent": Array [
              1,
              1,
              1,
            ],
            "start": Object {
              "column": 1,
              "line": 2,
              "offset": 1,
            },
          },
          "spread": false,
          "type": "listItem",
        },
      ],
      "ordered": true,
      "position": Position {
        "end": Object {
          "column": 7,
          "line": 5,
          "offset": 49,
        },
        "indent": Array [
          1,
          1,
          1,
        ],
        "start": Object {
          "column": 1,
          "line": 2,
          "offset": 1,
        },
      },
      "spread": false,
      "start": 1,
      "type": "list",
    },
    Object {
      "children": Array [
        Object {
          "type": "jsx",
          "value": "<CodeTabs>",
        },
        Object {
          "type": "jsx",
          "value": "<CodeBlock language=\\"plain\\" title=\\"\\" filename=\\"\\">",
        },
        Object {
          "lang": "plain",
          "meta": null,
          "position": Position {
            "end": Object {
              "column": 4,
              "line": 9,
              "offset": 76,
            },
            "indent": Array [
              1,
              1,
            ],
            "start": Object {
              "column": 1,
              "line": 7,
              "offset": 51,
            },
          },
          "type": "code",
          "value": "outside list",
        },
        Object {
          "type": "jsx",
          "value": "</CodeBlock>",
        },
        Object {
          "type": "jsx",
          "value": "</CodeTabs>",
        },
      ],
      "data": Object {
        "hName": "div",
        "hProperties": Object {
          "className": "code-tabs-wrapper",
        },
      },
      "lang": "plain",
      "meta": null,
      "position": Position {
        "end": Object {
          "column": 4,
          "line": 9,
          "offset": 76,
        },
        "indent": Array [
          1,
          1,
        ],
        "start": Object {
          "column": 1,
          "line": 7,
          "offset": 51,
        },
      },
      "type": "element",
      "value": "outside list",
    },
  ],
  "position": Object {
    "end": Object {
      "column": 1,
      "line": 10,
      "offset": 77,
    },
    "start": Object {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "type": "root",
}
`;

exports[`gatsby-plugin-code-tabs does not fold code blocks split by a paragraph 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [
        Object {
          "type": "jsx",
          "value": "<CodeTabs>",
        },
        Object {
          "type": "jsx",
          "value": "<CodeBlock language=\\"python\\" title=\\"\\" filename=\\"\\">",
        },
        Object {
          "lang": "python",
          "meta": null,
          "position": Position {
            "end": Object {
              "column": 4,
              "line": 4,
              "offset": 35,
            },
            "indent": Array [
              1,
              1,
            ],
            "start": Object {
              "column": 1,
              "line": 2,
              "offset": 1,
            },
          },
          "type": "code",
          "value": "print \\"Hello World!\\"",
        },
        Object {
          "type": "jsx",
          "value": "</CodeBlock>",
        },
        Object {
          "type": "jsx",
          "value": "</CodeTabs>",
        },
      ],
      "data": Object {
        "hName": "div",
        "hProperties": Object {
          "className": "code-tabs-wrapper",
        },
      },
      "lang": "python",
      "meta": null,
      "position": Position {
        "end": Object {
          "column": 4,
          "line": 4,
          "offset": 35,
        },
        "indent": Array [
          1,
          1,
        ],
        "start": Object {
          "column": 1,
          "line": 2,
          "offset": 1,
        },
      },
      "type": "element",
      "value": "print \\"Hello World!\\"",
    },
    Object {
      "children": Array [
        Object {
          "position": Position {
            "end": Object {
              "column": 16,
              "line": 6,
              "offset": 52,
            },
            "indent": Array [],
            "start": Object {
              "column": 1,
              "line": 6,
              "offset": 37,
            },
          },
          "type": "text",
          "value": "some text here:",
        },
      ],
      "position": Position {
        "end": Object {
          "column": 16,
          "line": 6,
          "offset": 52,
        },
        "indent": Array [],
        "start": Object {
          "column": 1,
          "line": 6,
          "offset": 37,
        },
      },
      "type": "paragraph",
    },
    Object {
      "children": Array [
        Object {
          "type": "jsx",
          "value": "<CodeTabs>",
        },
        Object {
          "type": "jsx",
          "value": "<CodeBlock language=\\"javascript\\" title=\\"\\" filename=\\"\\">",
        },
        Object {
          "lang": "javascript",
          "meta": null,
          "position": Position {
            "end": Object {
              "column": 4,
              "line": 10,
              "offset": 100,
            },
            "indent": Array [
              1,
              1,
            ],
            "start": Object {
              "column": 1,
              "line": 8,
              "offset": 54,
            },
          },
          "type": "code",
          "value": "console.log(\\"Hello World!\\");",
        },
        Object {
          "type": "jsx",
          "value": "</CodeBlock>",
        },
        Object {
          "type": "jsx",
          "value": "</CodeTabs>",
        },
      ],
      "data": Object {
        "hName": "div",
        "hProperties": Object {
          "className": "code-tabs-wrapper",
        },
      },
      "lang": "javascript",
      "meta": null,
      "position": Position {
        "end": Object {
          "column": 4,
          "line": 10,
          "offset": 100,
        },
        "indent": Array [
          1,
          1,
        ],
        "start": Object {
          "column": 1,
          "line": 8,
          "offset": 54,
        },
      },
      "type": "element",
      "value": "console.log(\\"Hello World!\\");",
    },
  ],
  "position": Object {
    "end": Object {
      "column": 1,
      "line": 11,
      "offset": 101,
    },
    "start": Object {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "type": "root",
}
`;

exports[`gatsby-plugin-code-tabs folds two code blocks 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [
        Object {
          "type": "jsx",
          "value": "<CodeTabs>",
        },
        Object {
          "type": "jsx",
          "value": "<CodeBlock language=\\"python\\" title=\\"\\" filename=\\"\\">",
        },
        Object {
          "lang": "python",
          "meta": null,
          "position": Position {
            "end": Object {
              "column": 4,
              "line": 4,
              "offset": 35,
            },
            "indent": Array [
              1,
              1,
            ],
            "start": Object {
              "column": 1,
              "line": 2,
              "offset": 1,
            },
          },
          "type": "code",
          "value": "print \\"Hello World!\\"",
        },
        Object {
          "type": "jsx",
          "value": "</CodeBlock>",
        },
        Object {
          "type": "jsx",
          "value": "<CodeBlock language=\\"javascript\\" title=\\"\\" filename=\\"\\">",
        },
        Object {
          "lang": "javascript",
          "meta": null,
          "position": Position {
            "end": Object {
              "column": 4,
              "line": 8,
              "offset": 83,
            },
            "indent": Array [
              1,
              1,
            ],
            "start": Object {
              "column": 1,
              "line": 6,
              "offset": 37,
            },
          },
          "type": "code",
          "value": "console.log(\\"Hello World!\\");",
        },
        Object {
          "type": "jsx",
          "value": "</CodeBlock>",
        },
        Object {
          "type": "jsx",
          "value": "</CodeTabs>",
        },
      ],
      "data": Object {
        "hName": "div",
        "hProperties": Object {
          "className": "code-tabs-wrapper",
        },
      },
      "lang": "python",
      "meta": null,
      "position": Position {
        "end": Object {
          "column": 4,
          "line": 4,
          "offset": 35,
        },
        "indent": Array [
          1,
          1,
        ],
        "start": Object {
          "column": 1,
          "line": 2,
          "offset": 1,
        },
      },
      "type": "element",
      "value": "print \\"Hello World!\\"",
    },
  ],
  "position": Object {
    "end": Object {
      "column": 1,
      "line": 9,
      "offset": 84,
    },
    "start": Object {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "type": "root",
}
`;

exports[`gatsby-plugin-code-tabs supports filenames 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [
        Object {
          "type": "jsx",
          "value": "<CodeTabs>",
        },
        Object {
          "type": "jsx",
          "value": "<CodeBlock language=\\"plain\\" title=\\"Hello\\" filename=\\"hello.txt\\">",
        },
        Object {
          "lang": "plain",
          "meta": "{tabTitle: Hello} {filename: hello.txt}",
          "position": Position {
            "end": Object {
              "column": 4,
              "line": 4,
              "offset": 59,
            },
            "indent": Array [
              1,
              1,
            ],
            "start": Object {
              "column": 1,
              "line": 2,
              "offset": 1,
            },
          },
          "type": "code",
          "value": "first",
        },
        Object {
          "type": "jsx",
          "value": "</CodeBlock>",
        },
        Object {
          "type": "jsx",
          "value": "<CodeBlock language=\\"plain\\" title=\\"Goodbye\\" filename=\\"goodbye.txt\\">",
        },
        Object {
          "lang": "plain",
          "meta": "{tabTitle: Goodbye} {filename: goodbye.txt}",
          "position": Position {
            "end": Object {
              "column": 4,
              "line": 8,
              "offset": 124,
            },
            "indent": Array [
              1,
              1,
            ],
            "start": Object {
              "column": 1,
              "line": 6,
              "offset": 61,
            },
          },
          "type": "code",
          "value": "second",
        },
        Object {
          "type": "jsx",
          "value": "</CodeBlock>",
        },
        Object {
          "type": "jsx",
          "value": "</CodeTabs>",
        },
      ],
      "data": Object {
        "hName": "div",
        "hProperties": Object {
          "className": "code-tabs-wrapper",
        },
      },
      "lang": "plain",
      "meta": "{tabTitle: Hello} {filename: hello.txt}",
      "position": Position {
        "end": Object {
          "column": 4,
          "line": 4,
          "offset": 59,
        },
        "indent": Array [
          1,
          1,
        ],
        "start": Object {
          "column": 1,
          "line": 2,
          "offset": 1,
        },
      },
      "type": "element",
      "value": "first",
    },
  ],
  "position": Object {
    "end": Object {
      "column": 1,
      "line": 9,
      "offset": 125,
    },
    "start": Object {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "type": "root",
}
`;

exports[`gatsby-plugin-code-tabs supports tab titles 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [
        Object {
          "type": "jsx",
          "value": "<CodeTabs>",
        },
        Object {
          "type": "jsx",
          "value": "<CodeBlock language=\\"plain\\" title=\\"Hello\\" filename=\\"\\">",
        },
        Object {
          "lang": "plain",
          "meta": "{tabTitle: Hello}",
          "position": Position {
            "end": Object {
              "column": 4,
              "line": 4,
              "offset": 37,
            },
            "indent": Array [
              1,
              1,
            ],
            "start": Object {
              "column": 1,
              "line": 2,
              "offset": 1,
            },
          },
          "type": "code",
          "value": "first",
        },
        Object {
          "type": "jsx",
          "value": "</CodeBlock>",
        },
        Object {
          "type": "jsx",
          "value": "<CodeBlock language=\\"plain\\" title=\\"Goodbye\\" filename=\\"\\">",
        },
        Object {
          "lang": "plain",
          "meta": "{tabTitle: Goodbye}",
          "position": Position {
            "end": Object {
              "column": 4,
              "line": 8,
              "offset": 78,
            },
            "indent": Array [
              1,
              1,
            ],
            "start": Object {
              "column": 1,
              "line": 6,
              "offset": 39,
            },
          },
          "type": "code",
          "value": "second",
        },
        Object {
          "type": "jsx",
          "value": "</CodeBlock>",
        },
        Object {
          "type": "jsx",
          "value": "</CodeTabs>",
        },
      ],
      "data": Object {
        "hName": "div",
        "hProperties": Object {
          "className": "code-tabs-wrapper",
        },
      },
      "lang": "plain",
      "meta": "{tabTitle: Hello}",
      "position": Position {
        "end": Object {
          "column": 4,
          "line": 4,
          "offset": 37,
        },
        "indent": Array [
          1,
          1,
        ],
        "start": Object {
          "column": 1,
          "line": 2,
          "offset": 1,
        },
      },
      "type": "element",
      "value": "first",
    },
  ],
  "position": Object {
    "end": Object {
      "column": 1,
      "line": 9,
      "offset": 79,
    },
    "start": Object {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "type": "root",
}
`;
